### ISSUE TITLE
Window Capture (Xcomposite) black screen (BadAlloc) (GLXBadPixmap)

### ISSUE THREAD

**Issue by Ancient8** [obsproject/obs-studio#3862]
## Platform
Package type: Snap and apt install obs-studio
Operating system and version: Ubuntu 20.04.1 LTS
OBS Studio version: 26.0.2

## Expected Behavior
Upon choosing a window for Window Capture it should render appropriately. 

## Current Behavior
It does work with most applications but it does not work with what appear to root focused applications (terminal, nautilus, disk utility, dconf, etc), although some like applications like Steam also doesn't work but I'm not sure if it also does something to the root. But either way I don't know if root permissions has anything to do with it (probably not). After several restarts OBS sometimes can render the Steam window properly but after closing it starts giving the error once again.

## Steps to Reproduce
I've recorded a video with several tests showing how it happens and also added additional information about my system and obs settings:
https://www.youtube.com/watch?v=BJBqDaF2fqs

I've forgotten to include the opening and exiting of OBS on the video, it shows a memory leak, not sure if it's relevant to this issue or even if it's normal. Here's a .txt file of it:
https://mega.nz/file/QI40iKRB#Npnb9UnNawZK7qJQjOGEiM53n03cJT1PfeehgsKpQ_4

[Main error] 
error: xcompcap: glXCreatePixmap failed: BadAlloc (insufficient resources for operation)

[Secondary error] 
error: xcompcap: cleanup glXReleaseTexImageEXT failed: GLXBadPixmap

Tested in both Snap and apt install obs-studio versions.

## Additional information
Sorry if sometimes I block the view during the youtube video with the OBS Window Capture, I'm very new to recording or streaming.

**Comment by kkartaltepe** [comment]
Chromeos is not supported.

**Comment by Ancient8** [comment]
I'm sorry, am I missing something? Where did I stated Chromeos?
What even is that and am I using it?

**Comment by Ancient8** [comment]
> Chromeos is not supported.

I'm sorry, am I missing something? Where did I state i was using Chromeos?
I am using Ubuntu 20.04.1

**Comment by kkartaltepe** [comment]
> I know in the official website it states - "Please note that OBS Studio is not fully working on Chrome OS and features like Screen and Window Capture do not work."
> But I'm just reporting this and sending the video in hopes it can help the devs in any way.

**Comment by Ancient8** [comment]
I do not understand your first response still, my issue is directed at Ubuntu 20.04.1
I just added that quote in the sense of that this is a very known bug among the Linux community and it hasn't been solved.
Is Ubuntu 20.04.1 somehow related to Chromeos?

**Comment by kkartaltepe** [comment]
This is also not our bug, at least until someone can provide evidence otherwise. Consider filing it against your glx provider, X11 implementation, or your compositor all of which participate in this call and in our experience changing any one of these typically fixes the issue.

**Comment by Ancient8** [comment]
Well this was a weird experience, I'm not even sure anymore if the youtube vid was even analyse or read the full post because how quickly the issue was closed and from the first responses.
But alright, I guess I'll go elsewhere.

**Comment by Ancient8** [comment]
https://obsproject.com/forum/threads/obs-xcomposite-problem.23652/
https://obsproject.com/forum/threads/ubuntu-16-04-xcompcap-glxcreatepixmap-failed-badalloc.64695/
https://obsproject.com/forum/threads/trying-to-create-a-window-capture-for-any-window-except-obs-fails-with-badalloc.111541/
https://obsproject.com/forum/threads/xcomposite-doesnt-capture-all-windows.61491/
https://www.reddit.com/r/archlinux/comments/dry5iz/obs_studio_nvidia_mutter_gnome_and_others_black/
https://obsproject.com/forum/threads/window-capture-black-screen.123791/

And this is just what I found for linux and there's probably more, there's threads for Windows which is fixed by using integrated graphics instead of the gpu (usually Nvidia drivers) which still remains a bug even in the Windows version.
So I tried changing the my drivers from Nvidia 455 to X.Org Nouveau Display drivers and it worked but of course that would make everything extremely sluggish and wouldn't be possible to play games with it properly, similar to changing to integrated graphics on windows, but this time it would be worse because i can't assign Nouveau to obs alone like Windows does.

After some inspection the bug also seems to persist in Gnome, so that makes sense that in my tests (shown in video) nautilus, terminal, disk utility, etc. would be the ones affected.
This bug with OBS has been around since at least ubuntu 16.04, maybe even in older versions and also in Windows for a long time.

This should be further investigated, for now i don't think I can do anything else, i've tested everything I found in forums.
If you wish this issue to remain closed, so be it.

**Comment by Fenrirthviti** [comment]
Given the information provided, while the issue might have been closed hastily I am not sure what we could do on our end. If there is a fundamental issue with compositor/glx/x11 that doesn't allow for this kind of capture to occur, I am not sure what the go-forward here is, as all the solutions I have seen were found in the environment, not in OBS.

I'll reopen this for now, but please don't expect rapid movement on it.

**Comment by Ancient8** [comment]
No other Source on OBS seems to be affected besides Window Capture (Xcompositor) 2, I have found a possible direct link between Nvidia drivers (in Gnome desktop environments) and Xcomposite 2.
X.org Nouveau drivers and Virtualbox (VMWare tools) drivers work perfectly with the Xcompositor, but both nvidia 455 and 450 tests didn't work. I don't know about even older versions but it has been reported that even on nvidia drivers 364.19, and mesa 11.2.1 didn't also work, so this origin bug is from long time ago.
Godspeed.

**Comment by kkartaltepe** [comment]
This bug was closed because none of this information is new, we have known it since people have reported it years ago. Its also again unlikely to be a bug in obs, which is why Its recommended to raise a report with people who actually can change and debug the code at fault or at least provide any insight into why it occurs.

If others want to keep it open thats fine, but I have removed references that are misleading and conflating this with other unrelated issues in the OP. Since you have been spamming this across all of our channels please also do not conflate this with the variety of other issues affecting other platforms.

**Comment by Ancient8** [comment]
I posted on reddit and discord once, that's not spamming.
Consider this closed then, thanks anyways and good luck with the future development.

**Comment by WHK102** [comment]
Same problem in Ubuntu 20.04 lts, can not select gnome terminal as xcomposite window source.

**Comment by qwqw330** [comment]
Exactly the same problem here. For some applications just a black screen.

Programs tested that don't work: gnome-terminal, gedit
Programs that work: hexchat, firefox
DE: gnome3
OS: Ubuntu 20.04

**Comment by Ancient8** [comment]
For those who still experience this issue still 3 years later after the original post:

If someone finds a solution or workaround, would be greatly appreciated if they were to post or be mentioned here!
Please be free to share any information that you might think it's useful or correlated to the issue at hand.

I will add here my main suspicious to the problem for those who aren't going to bother to read the whole thread:

> X.org Nouveau drivers and Virtualbox (VMWare tools) drivers work perfectly with the Xcompositor, but both nvidia 455 and 450 tests didn't work.

If someone uses nvidia drivers and they work with the Xcomposite widnows capture, please share the GPU model and the nvidia drivers versions.

> (...) there's threads for Windows which is fixed by using integrated graphics instead of the gpu (usually Nvidia drivers) (...) So I tried changing (...) my drivers from Nvidia 455 to X.Org Nouveau Display drivers and it worked but of course that would make everything extremely sluggish and wouldn't be possible to play games with it properly (...) because i can't assign Nouveau to OBS alone like Windows does.

If someone knows how to only assign X.Org Nouveau Display only for OBS in Ubuntu, please share, this would most likely be a workaround, since it appears to be some sort of incompatibility between Nvidia drivers and GNOME.

> After some inspection the bug also seems to persist in Gnome and with Nvidia drivers, so that makes sense that in my tests (shown in video) nautilus, terminal, disk utility, dconf, etc. would be the ones affected. Not sure why Steam is also affected though.

Now these are just all theories, the issue at hand is still unknown, godspeed!

**Comment by Ancient8** [comment]
I would appreciate this issue to be marked as Open since there are those who still have this issue 3yr later after the original post.

If someone finds a solution, workaround or any information that you might think it's useful or correlated to the issue at hand, would be greatly appreciated if they were to post or be mentioned here!

Now these are all just theories, my main suspicious to the problem for those who don't bother reading the whole thread:

> X.org Nouveau drivers and Virtualbox (VMWare tools) drivers work perfectly with the Xcompositor, but both nvidia 455 and 450 tests didn't work.

If someone uses nvidia drivers and they work with the Xcomposite widnows capture, please share the GPU model and the nvidia drivers versions.

> (...) there's threads for Windows which is fixed by using integrated graphics instead of the gpu (usually Nvidia drivers) (...) So I tried changing (...) my drivers from Nvidia 455 to X.Org Nouveau Display drivers and it worked but of course that would make everything extremely sluggish and wouldn't be possible to play games with it properly (...) because i can't assign Nouveau to OBS alone like Windows does.

If someone knows how to only assign X.Org Nouveau Display only for OBS in Ubuntu and be able to use the Nvidia GPU for other applications, please share, this would most likely be a workaround, since it appears to be some sort of incompatibility between Nvidia drivers and GNOME.

> After some inspection the bug also seems to persist in Gnome and with Nvidia drivers, so that makes sense that in my tests (shown in video) nautilus, terminal, disk utility, dconf, etc. would be the ones affected. Not sure why Steam is also affected though.

**Comment by Ancient8** [comment]
If someone finds a solution, workaround or any information that you might think it's useful or correlated to the issue at hand, would be greatly appreciated if they were to post or be mentioned here!

This thread ended up being a bit extensive to read, so I thought i would summarize my main suspicious to the problem for those who don't bother reading the whole thread:

> X.org Nouveau drivers and Virtualbox (VMWare tools) drivers work perfectly with the Xcompositor, but both nvidia 455 and 450 tests didn't work.

If someone uses nvidia drivers and they work with the Xcomposite widnows capture, please share the GPU model and the nvidia drivers versions.

> (...) there's threads for Windows which is fixed by using integrated graphics instead of the gpu (usually Nvidia drivers) (...) So I tried changing (...) my drivers from Nvidia 455 to X.Org Nouveau Display drivers and it worked but of course that would make everything extremely sluggish and wouldn't be possible to play games with it properly (...) because i can't assign Nouveau to OBS alone like Windows does.

If someone knows how to only assign X.Org Nouveau Display only for OBS in Ubuntu and be able to use the Nvidia GPU for other applications, please share, this would most likely be a workaround, since it appears to be some sort of incompatibility between Nvidia drivers and GNOME.

> After some inspection the bug also seems to persist in Gnome and with Nvidia drivers, so that makes sense that in my tests (shown in video) nautilus, terminal, disk utility, dconf, etc. would be the ones affected. Not sure why Steam is also affected though.

@kkartaltepe I would appreciate this issue to be marked as Open since there are those who still have this issue 3yr later after the original post.

**Comment by Ancient8** [comment]
If someone finds a solution, workaround or any information that you might find useful to share or correlated to the issue at hand, would be greatly appreciated if they were to post or be mentioned here!

This thread ended up being a bit extensive to read, so I thought i would summarize my main suspicious to the problem for those who don't bother reading the whole thread:

    X.org Nouveau drivers and Virtualbox (VMWare tools) drivers work perfectly with the Xcompositor, but both nvidia 455 and 450 tests didn't work.

If someone uses nvidia drivers and they work with the Xcomposite widnows capture, please share the GPU model and the nvidia drivers versions.

    (...) there's threads for Windows which is fixed by using integrated graphics instead of the gpu (usually Nvidia drivers) (...) So I tried changing (...) my drivers from Nvidia 455 to X.Org Nouveau Display drivers and it worked but of course that would make everything extremely sluggish and wouldn't be possible to play games with it properly (...) because i can't assign Nouveau to OBS alone like Windows does.

If someone knows how to only assign X.Org Nouveau Display only for OBS in Ubuntu and be able to use the Nvidia GPU for other applications, please share, this would most likely be a workaround, since it appears to be some sort of incompatibility between Nvidia drivers and GNOME.

    After some inspection the bug also seems to persist in Gnome and with Nvidia drivers, so that makes sense that in my tests (shown in video) nautilus, terminal, disk utility, dconf, etc. would be the ones affected. Not sure why Steam is also affected

@kkartaltepe , @Fenrirthviti  I would appreciate this issue to be marked as Open since there are those who still have this issue 3 years later after the original post and might have new useful information to add to the already known issue, if you think it's not worth it you can keep it closed. Thank you for you support either way!

**Comment by Ancient8** [comment]
If someone finds a solution, workaround or any information that you might find useful to share or correlated to the issue at hand, would be greatly appreciated if they were to post or be mentioned here!

This thread ended up being a bit extensive to read, so I thought i would summarize my main suspicious/theories to the problem for those who don't bother reading the whole thread:

> X.org Nouveau drivers and Virtualbox (VMWare tools) drivers work perfectly with the Xcompositor, but both nvidia 455 and 450 tests didn't work.

If someone uses nvidia drivers and they work with the Xcomposite widnows capture, please share the GPU model and the nvidia drivers versions.

> (...) there's threads for Windows which is fixed by using integrated graphics instead of the gpu (usually Nvidia drivers) (...) So I tried changing (...) my drivers from Nvidia 455 to X.Org Nouveau Display drivers and it worked but of course that would make everything extremely sluggish and wouldn't be possible to play games with it properly (...) because i can't assign Nouveau to OBS alone like Windows does.

If someone knows how to only assign X.Org Nouveau Display only for OBS in Ubuntu and be able to use the Nvidia GPU for other applications, please share, this would most likely be a workaround, since it appears to be some sort of incompatibility between Nvidia drivers and GNOME.

> After some inspection the bug also seems to persist in Gnome and with Nvidia drivers, so that makes sense that in my tests (shown in video) nautilus, terminal, disk utility, dconf, etc. would be the ones affected. Not sure why Steam is also affected

@kkartaltepe , @Fenrirthviti I would appreciate this issue to be marked as Open since there are those who still have this issue 3 years later after the original post and might have new useful information to add to the already known issue, if you think it's not worth it you can keep it closed. Thank you for you support either way!

**Comment by Fenrirthviti** [comment]
Until we have reliable replication steps that point to an issue within OBS, it is our current policy that we don't keep upstream issues open on our repository. If someone is able to provide that, please submit a new issue report and we can investigate accordingly.

**Comment by kkartaltepe** [comment]
This has been identified as a limitation of nvidias drivers. File bug reports to them if you want but no driver vendor cares about making changes to X11 (and didnt 3 years ago either), so I would expect nothing to change.

Instead you can use wayland in a few months once the explicit sync changes land and xwayland releases are cut (or in 18-24 months if you prefer to use debian/ubuntu).

**Comment by kryztoval** [comment]
I hate that this issue is closed as "Completed". I think it should be closed as something else. As external bug maybe.

Anyways, idk who may find this useful, but as @Ancient8 Suggested I just stumbled upon this issue after it was working for a while. I checked my apt logs and I found the combination that worked for me.
So on a "NVIDIA Corporation GA102 [GeForce RTX 3090] (rev a1)"
In "Debian 12 Bookworm" one of the drivers that works is "nvidia-driver-full/stable,now 535.247.01-1~deb12u1"

The 550 version breaking the Xcomposite thing in the same configuration.

Dear OBS team, while I know this is not an OBS bug or error, the error is obvious only in obs. And OBS provides no error, the log files look exactly the same when it works and when it doesn't work. I am not saying it is your job to solve this bug which clearly resides in an incompatibility in the nVidia drivers totally outside your purview but maybe there could be a way to alert the user to the Xcomposite hook failing and that the issue is due to some driver incompatibility?
